sale_partner_selectable_option/views/res_partner_view.xml

```diff
@@ -6,12 +6,12 @@
         <field name="model">res.partner</field>
         <field name="inherit_id" ref="base.view_partner_form" />
         <field name="arch" type="xml">
-            <xpath
-                expr="//page[@name='sales_purchases']//field[@name='user_id']"
-                position="after"
-            >
-                <field name="selectable_in_sales_orders" />
-            </xpath>
+            <group name="sale" position="inside">
+                <field
+                    name="selectable_in_sales_orders"
```

Comment:
Sorry for not checking before, but the field name is a bit convoluted. It can be `sale_selectable`.